File: Examples/VisualBasic/AsposePDF/Annotations/DeleteAllAnnotationsFromPage.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf
Namespace AsposePDF.Annotations
    Public Class DeleteAllAnnotationsFromPage
        Public Shared Sub Run()
            ' ExStart:DeleteAllAnnotationsFromPage
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Annotations()

            ' Open document
            Dim pdfDocument As New Document(dataDir & Convert.ToString("DeleteAllAnnotationsFromPage.pdf"))

            ' Delete particular annotation
            pdfDocument.Pages(1).Annotations.Delete()

            dataDir = dataDir & Convert.ToString("DeleteAllAnnotationsFromPage_out_.pdf")
            ' Save updated document
            pdfDocument.Save(dataDir)
            ' ExEnd:DeleteAllAnnotationsFromPage
            Console.WriteLine(Convert.ToString(vbLf & "All annotation from a page deleted successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace

File: Examples/CSharp/AsposePDF/Annotations/DeleteAllAnnotationsFromPage.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;

namespace CSharp.AsposePdf.Annotations
{
    public class DeleteAllAnnotationsFromPage
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Annotations();

            //open document
            Document pdfDocument = new Document(dataDir + "DeleteAllAnnotationsFromPage.pdf");

            //delete particular annotation
            pdfDocument.Pages[1].Annotations.Delete();

            //save updated document
            pdfDocument.Save(dataDir + "DeleteAllAnnotationsFromPage_out.pdf");

        }
    }
}